### DISCUSSION TITLE
join with where bug?

### DISCUSSION THREAD

**Discussion by SergeyBakev** [fnc12/sqlite_orm#915] [Q&A] [answered]
Hello, i create query.
`	auto cols = _storage.select(
				&GroupSong::Nm,
				inner_join<Song>(on(is_equal(&Song::Name, &GroupSong::Name))),
				where(is_equal(&GroupSong::groupName, groupId)) && greater_than(sub(&GroupSong::NA, &GroupSong::Used), 0));`
it genereated next string :
```sql
SELECT "GroupSong"."name" 
FROM 'GroupSong' 
INNER JOIN  'songs' 
ON ("songs"."name" = "GroupSong"."name")  
WHERE (("GroupSong"."groupId" = ?)) AND (("GroupSong"."count" - "GroupSong"."used") > ?)
```
but it can not be executed. Where contains extra "(". What am I doing wrong?

**Comment by fnc12** [comment]
Hello. Did you try executing the same query in any SQLite client?

**Reply by SergeyBakev** [reply]
What is mean "any SQLite client". I execute it without "(" and all was successful.

**Reply by fnc12** [reply]
Looks like serialization bug. I'll check it and fix it today. Thanks

**Reply by fnc12** [reply, marked as answer]
It is not a bug. You wrote the query incorrectly. You wrote `WHERE (...) AND A > B` but it doesn't work like that. You must write `WHERE (... AND A > B)` instead. You wrote:
```c++
auto cols = _storage.select(
        &GroupSong::Nm,
        inner_join<Song>(on(is_equal(&Song::Name, &GroupSong::Name))),
        where(is_equal(&GroupSong::groupName, groupId)) && greater_than(sub(&GroupSong::NA, &GroupSong::Used), 0));
```
but the correct version is
```c++
auto cols = _storage.select(
        &GroupSong::Nm,
        inner_join<Song>(on(is_equal(&Song::Name, &GroupSong::Name))),
        where(is_equal(&GroupSong::groupName, groupId) && greater_than(sub(&GroupSong::NA, &GroupSong::Used), 0)));
```

**Reply by SergeyBakev** [reply]
Sorry, I didn't see the error. Thanks
